db/migrate/20150514114044_migrate_ptables_to_templates.rb

```diff
@@ -0,0 +1,84 @@
+class MigratePtablesToTemplates < ActiveRecord::Migration
+  class FakeOldPtable < ActiveRecord::Base
+    self.table_name = 'ptables'
+
+    has_and_belongs_to_many :operatingsystems, :join_table => 'operatingsystems_ptables', :foreign_key => 'ptable_id'
+    has_many :hostgroups, :foreign_key => 'ptable_id'
+    has_many_hosts :foreign_key => 'ptable_id'
+  end
+
+  class FakeNewPtable < ActiveRecord::Base
+    self.table_name = 'templates'
+
+    has_and_belongs_to_many :operatingsystems, :join_table => 'operatingsystems_ptables', :foreign_key => 'ptable_id'
+    has_many :hostgroups, :foreign_key => 'ptable_id'
+    has_many_hosts :foreign_key => 'ptable_id'
+  end
+
+  def up
+    remove_foreign_key "operatingsystems_ptables", :name => "operatingsystems_ptables_ptable_id_fk"
```

Comment:
When we introduced FKs, MySQL users would typically use the 'mysql' adapter.  This adapter didn't support FKs, so either it or foreigner would silently ignore add_foreign_key calls.  Now everybody's on 'mysql2', we get issues during upgrades when we call remove_foreign_key as the FKs aren't actually in the DB.

The real bug is that nobody's tried to fix the inconsistencies, but for now, we should handle it being missing.